packages/select/src/common/itemListRenderer.ts

```diff
@@ -4,6 +4,23 @@
  * Licensed under the terms of the LICENSE file distributed with this project.
  */
 
+export enum QueryListActiveItemType {
+    CREATE = "create",
+    ITEM = "item",
+}
+
+export interface IQueryListActiveItem<T> {
+    type: QueryListActiveItemType;
+    item: T | null;
+}
+
+export function getActiveItem<T>(activeItem: IQueryListActiveItem<T> | null | undefined): T | null {
+    if (activeItem && activeItem.type === QueryListActiveItemType.ITEM) {
```

Comment:
Prefer explicit boolean expressions for clarity and to generally avoid possibly unintended ignoring of falsey values (like `""`, `[]`). `activeItem != null`.

Also, I'd just use a ternary here.

```tsx
return activeItem != null && activeItem.type === QueryListActiveItemType.ITEM
    ? activeItem.item
    : null;
```